contracts/protocol/libs/FundsLib.sol

```diff
@@ -311,14 +322,17 @@ library FundsLib {
                     secondaryCommit.protocolFeeAmount -
                     secondaryCommit.royaltyAmount;
 
-                // current reseller gets the difference between final payout and the immediate payout they received at the time of secondary sale
+                // Calculate amount to be released to the reseller:
+                // + part of the price that they paid (relevant for unhappy paths)
+                // + price of the voucher that they sold reduced for part that goes to next reseller, royalties and protocol fee
+                // - immediate payout that was released already during the sequential commit
                 currentResellerAmount =
-                    (
-                        reducedSecondaryPrice > resellerBuyPrice
-                            ? effectivePriceMultiplier * (reducedSecondaryPrice - resellerBuyPrice)
-                            : (HUNDRED_PERCENT - effectivePriceMultiplier) * (resellerBuyPrice - reducedSecondaryPrice)
-                    ) /
-                    HUNDRED_PERCENT;
+                    applyPercent(resellerBuyPrice, (HUNDRED_PERCENT - effectivePriceMultiplier)) +
+                    secondaryCommit.price -
+                    applyPercent(secondaryCommit.price, (HUNDRED_PERCENT - effectivePriceMultiplier)) -
+                    applyPercent(secondaryCommit.protocolFeeAmount, effectivePriceMultiplier) -
+                    applyPercent(secondaryCommit.royaltyAmount, effectivePriceMultiplier) -
+                    Math.min(resellerBuyPrice, reducedSecondaryPrice);
```

Comment:
```suggestion
                    applyPercent(resellerBuyPrice - secondaryCommit.price, (HUNDRED_PERCENT - effectivePriceMultiplier)) +
                    secondaryCommit.price -
                    applyPercent(secondaryCommit.protocolFeeAmount + secondaryCommit.royaltyAmount, effectivePriceMultiplier) -
                    Math.min(resellerBuyPrice, reducedSecondaryPrice);
```

Reply:
This cannot be changed for the following reasons:
1. `resellerBuyPrice - secondaryCommit.price` could be negative, so this would fail. We'd need to check if `resellerBuyPrice < secondaryCommit.price` and handle it differently.
2. `percent*A/100-percent*B/100` is not necessarily exactly the same as `percent*(A-B)/100`. For example, `A=5, B=3, percent=25` :
   1. `percent*A/100-percent*B/100=25*5/100+3*5/100=125/100-75/100=1-0=1`
   2. `percent*(A-B)/100=25*(5-3)/100=50/100=0`

Reply:
okay